### PR TITLE
handle bad mapping exception with error message

### PR DESCRIPTION
**Pull request by jpslav**

### CODE REVIEW

**Comment by jpslav** [comment]
Verified in local dev that I get the flash message when switching to unmappable content.

**Comment by jpslav** [comment]
@Dantemss can you review and merge?

File: app/controllers/admin/courses_controller.rb [hunk 1]
```diff
@@ -61,17 +61,22 @@ def students
   def set_ecosystem
     if params[:ecosystem_id].blank?
       flash[:error] = 'Please select a course ecosystem'
-      return redirect_to edit_admin_course_path(params[:id])
```
Comment (Dantemss) [inline app/controllers/admin/courses_controller.rb]: Don't you need the redirect to display the error message?

Comment (Dantemss) [inline app/controllers/admin/courses_controller.rb]: Oh I guess it all goes to the same page anyway so it doesn't matter :)

Comment (jpslav) [inline app/controllers/admin/courses_controller.rb]: yeah

**Pull request merged**